File: src/features/organizations/dto/update-organization-datastore.dto.ts
import { IsBoolean, IsIn, IsOptional, IsString } from 'class-validator';

export class UpdateOrganizationDatastoreDto {
  @IsOptional()
  @IsBoolean()
  useDedicatedDb?: boolean;

  @IsOptional()
  @IsIn(['shared', 'dedicated'])
  type?: 'shared' | 'dedicated';

  @IsOptional()
  @IsString()
  databaseUri?: string;

  @IsOptional()
  @IsString()
  datastoreUri?: string;

  @IsOptional()
  @IsString()
  databaseName?: string;

  @IsOptional()
  @IsIn(['shared', 'dedicated'])
  dataResidency?: 'shared' | 'dedicated';
}

File: spring-boot-api/src/main/java/com/mytegroup/api/dto/organizations/UpdateOrganizationDatastoreDto.java
package com.mytegroup.api.dto.organizations;

import com.mytegroup.api.entity.enums.organization.DataResidency;
import com.mytegroup.api.entity.enums.organization.DatastoreType;

public record UpdateOrganizationDatastoreDto(
    Boolean useDedicatedDb,
    DatastoreType type,
    String databaseUri,
    String datastoreUri,
    String databaseName,
    DataResidency dataResidency
) {
}



